test/client_tests/grpc_or_grpcweb_channel_grpc_test.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021, the gRPC project authors. Please see the AUTHORS file
+// for details. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+@TestOn('!browser')
+
+import 'package:grpc/grpc_or_grpcweb.dart';
+import 'package:grpc/grpc.dart';
+import 'package:test/test.dart';
+
+const host = 'example.com';
+const grpcPort = 9090;
+const grpcWebPort = 8080;
+
+void main() {
+  test('Channel on non-web uses gRPC ClientChannel with correct params', () {
+    final channel = GrpcOrGrpcWebClientChannel(
+      host: host,
+      grpcPort: grpcPort,
+      grpcWebPort: grpcWebPort,
+      secure: false,
+    );
+    expect(channel is ClientChannel, isTrue);
+    final grpcChannel = channel as ClientChannel;
+    expect(grpcChannel.host, equals(host));
+    expect(grpcChannel.port, equals(grpcPort));
+    expect(grpcChannel.options.credentials.isSecure, isFalse);
```

Comment:
I was considering to check also if all other options have default values, but not sure if it's a good idea: please advice.

Reply:
I think it is okay to just test a subset.